src/main/java/net/mcreator/ui/minecraft/states/block/JBlockPropertiesStatesList.java

```diff
@@ -0,0 +1,224 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2024, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.ui.minecraft.states.block;
+
+import net.mcreator.ui.MCreator;
+import net.mcreator.ui.component.JEmptyBox;
+import net.mcreator.ui.component.JMinMaxSpinner;
+import net.mcreator.ui.component.TechnicalButton;
+import net.mcreator.ui.component.entries.JEntriesList;
+import net.mcreator.ui.component.util.PanelUtils;
+import net.mcreator.ui.dialogs.MCreatorDialog;
+import net.mcreator.ui.help.HelpUtils;
+import net.mcreator.ui.help.IHelpContext;
+import net.mcreator.ui.init.L10N;
+import net.mcreator.ui.init.UIRES;
+import net.mcreator.ui.laf.themes.Theme;
+import net.mcreator.ui.minecraft.states.PropertyData;
+import net.mcreator.ui.minecraft.states.PropertyDataWithValue;
+import net.mcreator.ui.validation.AggregatedValidationResult;
+import net.mcreator.ui.validation.Validator;
+import net.mcreator.ui.validation.component.VTextField;
+import net.mcreator.ui.validation.validators.RegistryNameValidator;
+import net.mcreator.ui.validation.validators.UniqueNameValidator;
+import net.mcreator.workspace.elements.ModElement;
+
+import javax.swing.*;
+import java.awt.*;
+import java.util.List;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class JBlockPropertiesStatesList extends JEntriesList {
+
+	private final List<JBlockPropertiesListEntry> propertiesList = new ArrayList<>();
+
+	private final JPanel propertyEntries = new JPanel();
+
+	private final TechnicalButton addProperty = new TechnicalButton(UIRES.get("16px.add"));
+
+	private final Map<?, ?> blockBaseProperties;
+	private final List<String> forbiddenProperties = new ArrayList<>();
+	private String blockBase;
+
+	public JBlockPropertiesStatesList(MCreator mcreator, IHelpContext gui, ModElement modElement) {
+		super(mcreator, new BorderLayout(0, 10), gui);
+		this.blockBaseProperties = Objects.requireNonNullElse(
+				(Map<?, ?>) mcreator.getWorkspace().getGenerator().getGeneratorConfiguration().getDefinitionsProvider()
+						.getModElementDefinition(modElement.getType()).get("block_base_properties"),
+				Collections.emptyMap());
+		if (this.blockBaseProperties.get("_default") instanceof List<?> commonProps) {

```

Comment:
Just leaving this note here, so we can resolve it later

> I should be able to add property e.g. called "waterlogged" if waterlogging is not enabled, so instead of "_default", make validator populate forbidden names based on the actual UI state. I can see cases where one may implement custom axis or face property in the future using states system, but right now we prevent this from being possible. To avoid bugs at this point, keep in mind waterlogged can be checked but not enabled due to block base, so take this into account when fixing this system